src/frontends/android/app/src/main/java/org/strongswan/android/ipc/VpnProfileCrudMessengerService.java

```diff
@@ -20,33 +22,47 @@
 
     private static final String TAG = "VpnProfileService";
     private VpnProfileCrud vpnProfileCrud;
-
+    private CallerVerificator callerVerificator;
+    private int uid;
     Messenger messenger = new Messenger(new Handler() {
-
+        @Override
+        public boolean sendMessageAtTime(Message msg, long uptimeMillis) {
+            uid = Binder.getCallingUid();
+            return super.sendMessageAtTime(msg, uptimeMillis);
+        }
         @Override
         public void handleMessage(Message msg) {
-            if (msg.what == getInteger(R.integer.vpn_profile_create_message)) {
-                create(msg);
-            } else if (msg.what == getInteger(R.integer.vpn_profile_read_message)) {
-                read(msg);
-            } else if (msg.what == getInteger(R.integer.vpn_profile_read_all_message)) {
-                readAll(msg);
-            } else if (msg.what == getInteger(R.integer.vpn_profile_update_message)) {
-                update(msg);
-            } else if (msg.what == getInteger(R.integer.vpn_profile_delete_message)) {
-                delete(msg);
-            } else if (msg.what == getInteger(R.integer.vpn_profile_delete_all_message)) {
-                deleteAll(msg);
-            } else if (msg.what == getInteger(R.integer.vpn_profile_delete_profile_certificate_message)) {
-                deleteCertificate(msg);
-            } else if (msg.what == getInteger(R.integer.vpn_profile_read_by_name_message)) {
-                readByName(msg);
-            } else {
-                Log.w(TAG, "Unknown message: " + msg);
-                super.handleMessage(msg);
+            if ( callerVerificator.isCallerPermitted(uid)){
+
+                if (msg.what == getInteger(R.integer.vpn_profile_create_message)) {
+                    create(msg);
+                } else if (msg.what == getInteger(R.integer.vpn_profile_read_message)) {
+                    read(msg);
+                } else if (msg.what == getInteger(R.integer.vpn_profile_read_all_message)) {
+                    readAll(msg);
+                } else if (msg.what == getInteger(R.integer.vpn_profile_update_message)) {
+                    update(msg);
+                } else if (msg.what == getInteger(R.integer.vpn_profile_delete_message)) {
+                    delete(msg);
+                } else if (msg.what == getInteger(R.integer.vpn_profile_delete_all_message)) {
+                    deleteAll(msg);
+                } else if (msg.what == getInteger(R.integer.vpn_profile_delete_profile_certificate_message)) {
+                    deleteCertificate(msg);
+                } else if (msg.what == getInteger(R.integer.vpn_profile_read_by_name_message)) {
+                    readByName(msg);
+                } else {
+                    Log.w(TAG, "Unknown message: " + msg);
+                    super.handleMessage(msg);
+                }
+
+
+
+            }else{
+                Log.e(TAG, "Cannot verify caller  uid " +uid  );
```

Comment:
fix whitespace in this line:
- doublespace between caller and uid
- no space after + and uid
- too much space between uid and );